media-video/pipewire/pipewire-0.3.23.ebuild

```diff
@@ -0,0 +1,207 @@
+		-Dexamples=true # As of v0.3.23 still contains the pipewire-media-session binary
+		-Dmedia-session=true # Allegedly should provide pipewire-media-session, except not
+		-Dman=true
+		$(meson_use test tests)
+		$(meson_use debug installed_tests) # Installs tests for later use (NOP unless test USE flag is also set)
+		$(meson_use gstreamer)
+		$(meson_use gstreamer gstreamer-device-provider)
+		$(meson_use systemd)
+		-Dsystemd-system-service=false # Matches upstream
+		$(meson_use systemd systemd-user-service) # Technically against Gentoo guidelines but we already have the USE flag for libsystemd support
+		$(meson_use pipewire-alsa) # Allows integrating ALSA apps into PW graph
+		$(meson_use pipewire-jack) # Allow integrating JACK apps into PW graph
+#		-Dlibjack-path="" # Where to install the libjack.so library
+		-Dspa-plugins=true
+		$(meson_use alsa) # Allows using kernel ALSA for sound I/O
+		-Daudiomixer=true # Matches upstream
+		-Daudioconvert=true # Matches upstream
+		$(meson_use bluetooth bluez5)
+		$(meson_use bluetooth bluez5-backend-hsp-native)
+		$(meson_use bluetooth bluez5-backend-hfp-native)
+		$(meson_use bluetooth bluez5-backend-ofono)
+		-Dbluez5-backend-hsphfpd=false # Matches upstream
+		-Dcontrol=true # Matches upstream
+		$(meson_use debug audiotestsrc) # Generate an audio test pattern
+		$(meson_use ffmpeg-spa ffmpeg) # Presumably makes FFmpeg encoder and decoder nodes addable to the graph (probably no users right now)
+		$(meson_use jack) # Allows PW to act as a JACK client
+		-Dsupport=true # Miscellaneous/common plugins, such as null sink
+		-Devl=false # Matches upstream
+		$(meson_use test) # Fake sink and source SPA for running tests
+		$(meson_use v4l v4l2)
+		-Dlibcamera=false # libcamera is not in Portage
+		-Dvideoconvert=true # Matches upstream
+		$(meson_use debug videotestsrc) # Generate a video test pattern
+		-Dvolume=false # Matches upstream
+		$(meson_use vulkan-spa vulkan) # Uses pre-compiled Vulkan compute shader to provide a CGI video source (looks like a dev thing)
+		$(meson_use extra pw-cat)
+		$(meson_use udev)
+#		-Dudevrulesdir="" # Directory for udev rules (defaults to /lib/udev/rules.d)
+#		-Dsystemd-user-unit-dir="" # Directory for user systemd units (defaults to /usr/lib/systemd/user)
+		$(meson_feature debug sdl2) # Enables SDL2 dependent code (currently only examples when -Dinstalled_tests=true)
+		$(meson_feature extra sndfile) # Enables libsndfile dependent code (currently only pw-cat)
+	)
+	meson_src_configure
+}
+
+src_compile() {
+	meson_src_compile
+
+	einfo "Generating ${limitsdfile}"
+	cat > ${limitsdfile} <<- EOF || die
+		# Start of ${limitsdfile} from ${P}
+
+		1000:60000	-	memlock 256
+
+		# End of ${limitsdfile} from ${P}
+	EOF
+}
+
+src_install() {
+	meson_src_install
+
+	insinto /etc/security/limits.d/
+	doins ${limitsdfile}
+
+	if use pipewire-alsa; then
+		dosym ../../../usr/share/alsa/alsa.conf.d/50-pipewire.conf /etc/alsa/conf.d/50-pipewire.conf
```

Comment:
I'm not a dev, but are you sure it's a good idea to use an absolute path here?

Reply:
The only absolute path is _/etc/alsa/conf.d/50-pipewire.conf_ which I assume is kosher as every other ebuild I looked at is doing it that way. As for the relative path of _../../../usr/share/alsa/alsa.conf.d/50-pipewire.conf_, I did try replacing it with  _${EROOT}_ but **repoman full -d -x** was not happy about that, so I can only assume that dosym is inherently EPREFIX aware (which would explain why everyone is using plain _/etc/.._).

Reply:
Syntax is correct, without having checked actuall linkage here.

Reply:
As a quick heads up, I have since then replaced _dosym_ with _dosym8 -r_ which allows to pass in presumably prefix aware absolute paths and get a relative link to the target as a result.